File: Michael_Hug_CS4150_Project_spr2014/Java/src/globals/Memory.java
/*
 *Author : Michael Hug
 *Author email : hmichae4@students.kennesaw.edu
 *Student of Prof Gayler cs4150 Spr014
 *Project - Java
 */
package globals;

import exceptions.UndefinedVariable;
import lexicalanalyzer.Token;

public class Memory
{

    private static final Integer[] mem = new Integer[52];

    /**
     * @param token
     * @return value stored for specified variable
     * @throws exceptions.UndefinedVariable
     */
    public static int fetch(Token token) throws UndefinedVariable
    {
        Integer intgr = mem[getArrayIndex(token.getCharacter())];
        if(intgr==null)
            throw new UndefinedVariable(token);
        return intgr;
    }

    /**
     * postcondition: value has been stored as the value of the specified variable
     * @param token
     * @param value
     */
    public static void store(Token token, int value)
    {
        mem[getArrayIndex(token.getCharacter())] = value;
    }
    private static int getArrayIndex(char minuend)
    {
        int subtrahend = (minuend<91) ? 65 : 97;
        return minuend - subtrahend;
    }
}

File: Michael_Hug_CS4150_Project_spr2014/python/src/globals/memory.py
mem = [52]

def set(t,i):
    mem[getArrayIndex(t)]=i

def get(t):
    if(t.getCharacter()==null):
        print("integer token expected",t)
        sys.exit(1)
    i = mem[getArrayIndex(t)]
    if(i==null):
        print("undefined variable",t)
        sys.exit(2)
    return i;
    
#A=65 , Z=90 , a = 97 , z = 122
def getArrayIndex(t):
    index = t.getCharacter();
    if(index<91):
        index-=65;
    else:
        index-=97;
    return index;